src/Microsoft.AspNet.Http.Core/RequestCookiesFeature.cs

```diff
@@ -44,18 +44,23 @@ public IReadableStringCollection Cookies
                 }
 
                 var headers = _request.Fetch(_features).Headers;
-                string cookiesHeader = ParsingHelpers.GetHeader(headers, Constants.Headers.Cookie) ?? string.Empty;
+                string[] values;
+                if (!headers.TryGetValue(HeaderNames.Cookie, out values))
+                {
+                    values = new string[0];
+                }
+                var cookiesHeader = string.Join(", ", values);
```

Comment:
We use it to check for changes next time you access the collection. Doing Equals on a string[] is harder, and you never know if someone is going to change your own array references.  If it shows up on perf we could reduce it to a fancy hash code check.